sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -182,6 +183,10 @@ case class DataSource(
       throw new AnalysisException(
         s"Unable to infer schema for $format. It must be specified manually.")
     }
+
+    SchemaUtils.checkSchemaColumnNameDuplication(
+      dataSchema, "the datasource", sparkSession.sessionState.conf.caseSensitiveAnalysis)
```

Comment:
`dataSchema + partitionSchema` is our full schema. When checking duplication, we need to check the full schema together. 

Could you also add such a test case?